components/script/dom/htmlformcontrolscollection.rs

```diff
@@ -43,8 +47,40 @@ impl HTMLFormControlsCollection {
 
 impl HTMLFormControlsCollectionMethods for HTMLFormControlsCollection {
     // https://html.spec.whatwg.org/multipage/#dom-htmlformcontrolscollection-nameditem
-    fn NamedGetter(&self, name: DOMString, found: &mut bool) -> Option<Root<Element>> {
-        self.collection.NamedGetter(name, found)
+    fn NamedItem(&self, name: DOMString) -> Option<RadioNodeListOrElement> {
+        // Step 1
+        if name.is_empty() { return None; }
+
+        let count = self.collection.elements_iter().filter_map(|elem| {
+            if elem.get_string_attribute(&atom!("name")) == name
+               || elem.get_string_attribute(&atom!("id")) == name {
+                Some(elem)
+            } else { None }
```

Comment:
You could alternatively just return `1` or `0` in here and then just call `sum()` on the iterator 

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.sum